dataTrieMigrator/dataTrieMigrator.go

```diff
@@ -0,0 +1,82 @@
+package dataTrieMigrator
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+type dataTrieMigrator struct {
```

Comment:
So this is only the wrapper between the builtinfunctions system and the account. By using interfaces, the code is well-decoupled. :+1: